src/components/Reports/ReportFormAnnual.tsx

```diff
@@ -199,11 +254,42 @@ export default function ReportFormAnnual(props: ReportFormAnnualProps): JSX.Elem
           }}
         />
       </Grid>
-      <Grid item xs={12}>
+      <Grid item xs={12} ref={divRef}>
+        <input type='file' ref={inputRef} className={classes.hiddenInput} value='' onChange={onFileChosen} />
         <Typography fontSize='14px' fontWeight={400} color={theme.palette.TwClrTxtSecondary}>
           {strings.BUDGET_DOCUMENT_XLS}
         </Typography>
-        <Button onClick={handleFileUploadButtonClick} priority='secondary' type='passive' label={strings.CHOOSE_FILE} />
+        <Button
+          disabled={!editable}
+          onClick={onChooseFileHandler}
+          priority='secondary'
+          type='passive'
+          label={strings.CHOOSE_FILE}
+        />
+        {newFiles.map((f) => (
+          <Box key={f.name} display='flex' alignItems='center'>
+            <Button
+              disabled={!editable}
+              priority='ghost'
+              type='passive'
+              icon='iconTrashCan'
+              onClick={() => onRemoveFile(f.name)}
+            />
+            <Typography>{f.name}</Typography>
+          </Box>
+        ))}
+        {existingFiles.map((f) => (
+          <Box key={f.id} display='flex' alignItems='center'>
+            {editable ? (
+              <Button priority='ghost' type='passive' icon='iconTrashCan' onClick={() => onRemoveExistingFile(f.id)} />
+            ) : (
+              <a href={getReportFileUrl(report.id, f.id)}>
```

Comment:
Not sure how permissions for this report file url works,, guessing it isn't public. Check with @sgrimm ?
For authenticated routes, may want to issue a GET within this session and make it downloadable as a data blob.

Reply:
Not 100% sure what the question is here, but all the `/api/v1/reports` endpoints require authentication and enforce the usual permissions. The GET endpoint for retrieving files sets the `Content-Disposition` header to `attachment` with a filename so the browser won't display it inline.

Reply:
Server side of this, for reference: https://github.com/terraware/terraware-server/blob/main/src/main/kotlin/com/terraformation/backend/report/api/ReportsController.kt#L255

Reply:
I thought the `href=<authenticated-route>` link/button would not work but it actually does since we use cookies and same url host for API. My question about permissions is no longer an issue. 
Glad I asked this question though, the `Content-Disposition` needs to be considered.

Charlie, if it helps -  [blob](https://developer.mozilla.org/en-US/docs/Web/API/Blob/Blob) 
Setting an href with blob data.. https://javascript.info/blob#blob-as-url

Reply:
I'd assumed we'd want to do more or less what this PR does, where the browser treats the file as a download and the React code never touches the file contents. If instead there's an advantage to loading the file into memory in JavaScript and then generating a copy of it for download, we can easily get rid of the `Content-Disposition` header.

Reply:
I think we're good to go with regards to the `Content-Disposition` it causes the file to be downloaded as an attachment as desired
<img width="768" alt="image" src="https://user-images.githubusercontent.com/114949086/224166104-d1375a3b-773e-4459-81d7-2cf6ef5639e0.png">
I'll take a look at the Blob things

Reply:
Thanks for the clarification and answering my questions, makes sense to me now. Looks good then 👍

Reply:
Yup I think leaving as-is is the way to go here.